connectors/sources/onedrive.py

```diff
@@ -535,23 +558,25 @@ async def get_docs(self, filtering=None):
 
                 for mail in user_mails:
                     user_id = user_mail_id_map.get(mail)
-                    async for entity in self.client.get_owned_files(
+                    async for entity, download_url in self.client.get_owned_files(
                         user_id, skipped_extensions, pattern
                     ):
                         entity = self.prepare_doc(entity)
                         if entity["type"] == FILE:
                             yield entity, partial(
-                                self.get_content, entity.copy(), user_id
+                                self.get_content, entity.copy(), download_url
```

Comment:
`download_url` can be easily checked here, instead of yielding it blindly. 

Although it returns immediately in `get_content`, it will still be wrapped and executed as a task, which is not necessary.